core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -446,6 +447,64 @@ def validate_funding_profiles_funding_secured_not_null(
         ]
 
 
+def validate_psi_funding_not_negative(
+    workbook: dict[str, pd.DataFrame]
+) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """
+    Validates that Private Sector Funding amounts are not negative.
+
+    This function checks the 'Private Investments' sheet for rows where either
+    'Private Sector Funding Required' or 'Private Sector Funding Secured' is negative.
+    If invalid rows are found, it returns a list of validation failures.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors or None if no errors are found.
+    """
+    psi_df = workbook["Private Investments"]
+    psi_df["Private Sector Funding Required"] = pd.to_numeric(
+        psi_df["Private Sector Funding Required"], errors="coerce"
+    )
```

Comment:
can you add a comment explaining what this is doing, including errors="coerce"

It also looks like this is modifying the source data - are we sure we want to do that? Can we instead pull it out as a new series, create the mask on that, and then use the mask to filter rows in the original df

Reply:
Added, and made it so source data is unmodified